### DISCUSSION TITLE
Target class [csp-nonce] does not exist.

### DISCUSSION THREAD

**Discussion by bluetiger1008** [spatie/laravel-csp#111] [Q&A]
`->addNonceForDirective(Directive::SCRIPT);` is causing this error on the server:
![Screenshot 2023-01-03 at 3 27 59 AM](https://user-images.githubusercontent.com/17339444/210322176-fb47c4c2-e577-4ac2-ab89-9daf3c7268fe.png)

On my local, I don't see this error, please help me resolve this.

**Comment by tegarkurniawan** [comment]
same issue please help me

**Comment by Rah1x** [comment]
This is because you didnt add the service provider to the laravel's app.php.

1) Either add the following:
`\Spatie\Csp\CspServiceProvider` 

2) Or, dont use `addNonceForDirective()`
Basically, if you are creating a custom policy, dont extend the default shipped `Basic` policy as that calls addNonceForDirective. Instead directly extend `Policy` and copy the code from Basic policy minus the addNonceForDirective part.

**Comment by cwilsn** [comment]
Make sure you have `nonce_enabled` set to true in your csp config.
